auth/mockServer.go

```diff
@@ -124,37 +140,68 @@ func (m *MockServer) SignTokenWithAdditionalClaims(claims OIDCClaims, additional
 		}
 		mapClaims[k] = v
 	}
+	token := jwt.New()
+
+	for k, v := range mapClaims {
+		token.Set(k, v)
+	}
+
+	return m.signToken(token, header)
+}
+
+func (m *MockServer) signToken(token jwt.Token, header map[string]interface{}) (string, error) {
+	jwkKey, err := jwk.New(m.RSAKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to create JWK: %s\n", err)
+	}
+	jwkKey.Set(jwk.KeyIDKey, header[propKid])
 
-	token := &jwtgo.Token{
-		Header: header,
-		Claims: mapClaims,
-		Method: jwtgo.SigningMethodRS256,
+	signedJwt, err := jwt.Sign(token, jwa.RS256, jwkKey)
+	if err != nil {
+		return "", fmt.Errorf("failed to sign the token: %v", err)
+	}
+
+	var alg, ok = header[propAlg].(jwa.SignatureAlgorithm)
```

Comment:
isn't it possible to check and adapt "wrong" `alg` header before signing the token?

Reply:
=> it isn't as jwt.Sign does not accept algorithm jwa.NoSignature or jwa.HS256.